The one failure looks to me like a flake or unrelated issue?

```
FAILED providers/docker/tests/unit/docker/decorators/test_docker.py::TestDockerDecorator::test_failing_task - docker.errors.APIError: 500 Server Error for http+docker://localhost/v1.48/images/create?tag=3.9-slim&fromImage=python: Internal Server Error ("toomanyrequests: You have reached your unauthenticated pull rate limit. https://www.docker.com/increase-rate-limit")
```

--- a/pyproject.toml
+++ b/pyproject.toml
@@ -562,11 +562,9 @@ bypass-selection = true
 # Needed until https://github.com/astral-sh/ruff/issues/8237 is available.
 [tool.black]
 line-length = 110
-target-version = ['py310', 'py311', 'py312']
 
 ## ruff settings ##
 [tool.ruff]
-target-version = "py310"
 line-length = 110
 extend-exclude = [
     ".eggs",

